src/control/lib/control/pool.go

```diff
@@ -1227,105 +1227,139 @@ func ListPools(ctx context.Context, rpcClient UnaryInvoker, req *ListPoolsReq) (
 
-			// DAOS-10885: Initialize the marking board with all the usable ranks
-			ranksWithoutSmd[scmNamespace.Mount.Rank] = scmNamespace
+			log.Debugf("Added SMD device %s (instance %d, ctrlr %s) is usable: device state=%q, smd-size=%d ctrlr-total-free=%d",
+				smdDevice.UUID, smdDevice.Rank, smdDevice.TrAddr, smdDevice.NvmeState.String(),
+				smdDevice.AvailBytes, rankNVMeFreeSpace[smdDevice.Rank])
 		}
+	}
 
-		nvmeRanksBytes := make(map[system.Rank]uint64, 0)
-		for _, nvmeController := range hostStorage.NvmeDevices {
-			for _, smdDevice := range nvmeController.SmdDevices {
-				if !smdDevice.NvmeState.IsNormal() {
-					log.Noticef("SMD device %s (instance %d, ctrlr %s) "+
-						"not usable (device state %q)",
-						smdDevice.UUID, smdDevice.Rank, smdDevice.TrAddr,
-						smdDevice.NvmeState.String())
-					continue
-				}
-
-				if len(ranks) != 0 && !smdDevice.Rank.InList(ranks) {
-					log.Debugf("Skipping SMD device %s (instance %d, ctrlr %s, rank %s) not in ranklist: %s",
-						smdDevice.UUID, smdDevice.Rank, smdDevice.TrAddr, smdDevice.Rank, ranks)
-					continue
-				}
-
-				nvmeRanksBytes[smdDevice.Rank] += smdDevice.AvailBytes
-				log.Debugf("Adding SMD device %s (instance %d, ctrlr %s) is usable: device state=%q, size=%d total=%d",
-					smdDevice.UUID, smdDevice.Rank, smdDevice.TrAddr,
-					smdDevice.NvmeState.String(), smdDevice.AvailBytes,
-					nvmeRanksBytes[smdDevice.Rank])
-			}
+	return nil
+}
+
+// Return the maximal SCM and NVMe size of a pool which could be created with all the storage nodes.
+func GetMaxPoolSize(ctx context.Context, log logging.Logger, rpcClient UnaryInvoker, ranks []system.Rank) (uint64, uint64, error) {
+	resp, err := StorageScan(ctx, rpcClient, &StorageScanReq{Usage: true})
+	if err != nil {
+		return 0, 0, err
+	}
+
+	if len(resp.HostStorage) == 0 {
+		return 0, 0, errors.New("empty host storage response from StorageScan")
+	}
+
+	// Generate function to verify a rank is in the provided rank slice.
+	checkRank := newCheckRankFunc(ranks)
+
+	rankNVMeFreeSpace := make(rankFreeSpaceMap)
+	scmBytes := uint64(math.MaxUint64)
+
+	for _, key := range resp.HostStorage.Keys() {
+		hostStorage := resp.HostStorage[key].HostStorage
+
+		if hostStorage.ScmNamespaces.Free() == 0 {
+			return 0, 0, errors.Errorf("Host without SCM storage: hostname=%s",
+				resp.HostStorage[key].HostSet.String())
 		}
-		for rank, nvmeRankBytes := range nvmeRanksBytes {
-			if nvmeBytes > nvmeRankBytes {
-				nvmeBytes = nvmeRankBytes
-			}
 
-			if _, ok := ranksWithoutSmd[rank]; ok {
-				// DAOS-10885: Remove the usable rank with NVME storage from the
-				// marking board
-				delete(ranksWithoutSmd, rank)
-			}
+		sBytes, err := processSCMSpaceStats(log, checkRank, hostStorage.ScmNamespaces, rankNVMeFreeSpace)
+		if err != nil {
+			return 0, 0, err
+		}
+
+		if scmBytes > sBytes {
+			scmBytes = sBytes
+		}
+
+		// FIXME: Should we fail here if any of the namespaces have no free space
```

Comment:
This case is managed by the dmg command itself
https://github.com/daos-stack/daos/blob/7440dde89bf57ad2ea040d48bc18c44b09627488/src/control/cmd/dmg/pool.go#L150

As this function is also used by `dmg system storage usage`, an error should not be send at this level.

Reply:
ok, let's remove this comment then